magik-lint/src/main/java/nl/ramsolutions/sw/magik/lint/MagikLint.java

```diff
@@ -106,14 +107,51 @@ void showChecks(final Writer writer, final boolean showDisabled)
         continue;
       }
 
-      for (final MagikCheckHolder.Parameter parameter : holder.getParameters()) {
+      showParameters(writer, holder, showDefaultValues);
+    }
+  }
+
+  /**
+   * Show parameters for a check.
+   *
+   * @param writer Writer Write to write output to.
+   * @param showDefaultValues boolean Boolean to show default values for parameters or not.
+   * @throws ReflectiveOperationException -
+   * @throws IOException -
+   */
+  private void showParameters(
+      final Writer writer, final MagikCheckHolder holder, final boolean showDefaultValues)
+      throws ReflectiveOperationException, IOException {
+    for (final MagikCheckHolder.Parameter parameter : holder.getParameters()) {
+      if (!parameter.isExplicitlySet() && !showDefaultValues) {
```

Comment:
So if no value is set, and the `--with-default-values` is not given, the parameter will not be shown? That seems a bit confusing